### DISCUSSION TITLE
Is it possible to scale widgets with window size?

### DISCUSSION THREAD

**Discussion by terrarier2111** [hecrj/iced#1948] [Q&A]
There are size functions e.g on the Text widget, but the size is in pixels which means the widget will always be the same size no matter the window size, would it be possible for the text to be e.g 20% of the window's size consistently no matter the actual window size? Also would it be possible to offset widgets from the borders of the window by x% of the window's x size or y size?

**Comment by lufte** [comment]
For widgets you can use all the options of https://docs.rs/iced/latest/iced/enum.Length.html, some of them being relative to the size of the container. I'm not sure you can use them for text though.

The other thing you want is padding. Some widgets like Row, Column and Container support it. I don't know if they support the relative Length variants, but you can try.

**Reply by terrarier2111** [reply]
First of all sorry for the late response from my experiments the FillPortion variant which is the variant supporting relative sizes doesn't support percentages but just supports fractions of 1/n which could work but isn't exactly what i need. The padding on the other hand just allows for passing in Into<Pixels> which clearly shows that there is no way to make this relative to anything and its always just an absolute value of pixels. Is there a way to get the screensize from within the view method of the application? Because if so i could do the relative to absolute calculations myself. Also i didn't find a way yet to offset a widget from one side of the screen by a specific amount is there a way to do so?

**Reply by lufte** [reply]
FillPortion doesn't work with fractions of 1. According to the documentation (https://docs.rs/iced/latest/iced/enum.Length.html) it works by adding all the portions and considering that the total. But in any case; percentage, fractions of 1, or arbitrary totals, the end result is the same one. If you want percentages then just make sure the sum of all your portions is 100 and that's it.

Regarding padding you are correct, it seems to be absolute only.

As for offsetting a widget, what I have done in the past is using a combination of `iced::widget::Space` with `FillPortion`, basically adding an empty space with the length I want.

**Reply by skutcher** [reply]
just for the case you want to scale in general overwrite this function so that it returns the scale factor u like to have
https://docs.rs/iced/latest/iced/application/trait.Application.html#method.scale_factor
